### PR TITLE
Optional Api keys on the fly

### PR DESCRIPTION
**Pull request by dmoralesl**
NOT TESTED

Related with doofinder/dfchat#555.

### CODE REVIEW

File: lib/llm_composer.ex [hunk 1]
```diff
@@ -78,15 +78,16 @@ defmodule LlmComposer do
   ## Returns
     - A tuple containing `:ok` with the response or `:error` if the model call fails.
   """
-  @spec run_completion(Settings.t(), messages(), LlmResponse.t() | nil) :: Helpers.action_result()
-  def run_completion(settings, messages, previous_response \\ nil) do
+  @spec run_completion(Settings.t(), messages(), LlmResponse.t() | nil, binary | nil) ::
+          Helpers.action_result()
+  def run_completion(settings, messages, previous_response \\ nil, api_key \\ nil) do
```
Comment (hectorperez) [inline lib/llm_composer.ex:83]: Why don't we pass the optional api_key inside settings?

Comment (victor23k) [reply lib/llm_composer.ex:83]: +1

Comment (dmoralesl) [reply lib/llm_composer.ex]: Done! Not tested tho

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,6 +6,9 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 
 ## [Unreleased]
 
+## [0.3.4] - 2024-12-05
+- **OpenAI API Keys**: Load API keys on the fly when calling chat_completion. The param is sent inside the settings.
```
Comment (hectorperez) [inline CHANGELOG.md:10]:
```suggestion
- **OpenAI API Keys**: Use an API key passed as a parameter when calling chat_completion — overriding the global API key defined in the config. The param is sent inside the settings. 
```

File: lib/llm_composer/models/open_ai.ex [hunk 3]
```diff
@@ -43,9 +43,10 @@ defmodule LlmComposer.Models.OpenAI do
   """
   def run(messages, system_message, opts) do
     model = Keyword.get(opts, :model)
+    api_key = Keyword.get(opts, :model) || get_key()
```
Comment (hectorperez) [inline lib/llm_composer/models/open_ai.ex:46]:
```suggestion
    api_key = Keyword.get(opts, :api_key) || get_key()
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -32,6 +35,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 
 ---
 [Unreleased]: https://github.com/doofinder/llm_composer/compare/0.3.3...HEAD
+[0.3.4]: https://github.com/doofinder/llm_composer/compare/0.3.2...0.3.3
```
Comment (hectorperez) [inline CHANGELOG.md:38]:
```suggestion
[0.3.4]: https://github.com/doofinder/llm_composer/compare/0.3.3...0.3.4
```

**Review by hectorperez — CHANGES_REQUESTED** [review]

**Review by hectorperez — APPROVED** [review]
:rocket:

**Review by mmacia — APPROVED** [review]

**Pull request merged**